src/Datasource/EntityTrait.php

```diff
@@ -418,6 +418,36 @@ public function hiddenProperties($properties = null)
     }
 
     /**
+     * Sets hidden properties.
+     *
+     * @param null|array $properties Either an array of properties to treat as virtual or null to get properties
+     * @param bool $merge Merge the new properties with the existing. By default false.
+     * @return self
```

Comment:
I think it follows the recent discussion about self/static or $this, and self is basically a new instance or clean, whereas $this means the chained object instance itself again.